src/libraries/Microsoft.Extensions.Options.ConfigurationExtensions/tests/OptionsBuidlerConfigurationExtensionsTests.cs

```diff
@@ -102,5 +105,40 @@ public static void BindConfiguration_UsesConfigurationRootIfSectionNameIsEmptySt
 
             Assert.Equal(messageValue, options.Message);
         }
+
+        [Fact]
+        public static void BindConfiguration_UpdatesOptionOnConfigurationUpdate()
+        {
+            const string messageValue1 = "This is a test";
+            const string messageValue2 = "This is the message after update";
+
+            FakeConfigurationSource configSource = new()
+            {
+                InitialData = new Dictionary<string, string>
+                {
+                    [nameof(FakeOptions.Message)] = messageValue1,
+                }
+            };
+
+            var services = new ServiceCollection();
+            services.AddSingleton<IConfiguration>(new ConfigurationBuilder()
+                    .Add(configSource)
+                    .Build());
+            var optionsBuilder = services.AddOptions<FakeOptions>();
```

Comment:
```suggestion
            OptionsBuilder<FakeOptions> optionsBuilder = services.AddOptions<FakeOptions>();
```

nit on `var` as per [coding-style](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md): 
`We only use var when it's obvious what the variable type is (e.g. var stream = new FileStream(...) not var stream = OpenStandardInput()).`